tests/test_helpers/constants.ts

```diff
@@ -43,3 +45,140 @@ export const TEST_CHART_DATA = {
+const TEST_COMMANDS_PARTIAL: Partial<CommandMapping> = {
+  UPDATE_CELL: {
+    type: "UPDATE_CELL",
+    col: 0,
+    row: 0,
+    content: "hello",
+    sheetId: "sheetId",
+  },
+  CLEAR_CELL: {
+    type: "CLEAR_CELL",
+    col: 0,
+    row: 0,
+    sheetId: "sheetId",
+  },
+  DELETE_CONTENT: {
+    type: "DELETE_CONTENT",
+    target: target("A1"),
+    sheetId: "sheetId",
+  },
+  ADD_MERGE: {
+    type: "ADD_MERGE",
+    target: target("A1:A2"),
+    sheetId: "sheetId",
+  },
+  REMOVE_MERGE: {
+    type: "REMOVE_MERGE",
+    target: target("A1:A2"),
+    sheetId: "sheetId",
+  },
+  SET_FORMATTING: {
+    type: "SET_FORMATTING",
+    target: target("A1"),
+    style: { bold: true },
+    sheetId: "sheetId",
+  },
+  CLEAR_FORMATTING: {
+    type: "CLEAR_FORMATTING",
+    target: target("A1"),
+    sheetId: "sheetId",
+  },
+  SET_BORDER: {
+    type: "SET_BORDER",
+    col: 0,
+    row: 0,
+    border: { top: DEFAULT_BORDER_DESC },
+    sheetId: "sheetId",
+  },
+  CREATE_FILTER_TABLE: {
+    type: "CREATE_FILTER_TABLE",
+    target: target("A1"),
+    sheetId: "sheetId",
+  },
+  REMOVE_FILTER_TABLE: {
+    type: "REMOVE_FILTER_TABLE",
+    target: target("A1"),
+    sheetId: "sheetId",
+  },
+  HIDE_SHEET: {
+    type: "HIDE_SHEET",
+    sheetId: "sheetId",
+  },
+  CREATE_SHEET: {
+    type: "CREATE_SHEET",
+    sheetId: "newSheetId",
+    name: "newSheetName",
+    position: 0,
+  },
+  DUPLICATE_SHEET: {
+    type: "DUPLICATE_SHEET",
+    sheetId: "sheetId",
+    sheetIdTo: "duplicateSheetId",
+  },
+  ADD_CONDITIONAL_FORMAT: {
+    type: "ADD_CONDITIONAL_FORMAT",
+    ranges: toRangesData("sheetId", "A1"),
+    cf: {
+      id: "cfId",
+      rule: {
+        values: ["1"],
+        operator: "Equal",
+        type: "CellIsRule",
+        style: { fillColor: "#FF0000" },
+      },
+    },
+    sheetId: "sheetId",
+  },
+  CREATE_FIGURE: {
+    type: "CREATE_FIGURE",
+    figure: {
+      id: "figureId",
+      tag: "tag",
+      x: 0,
+      y: 0,
+      width: 100,
+      height: 100,
+    },
+    sheetId: "sheetId",
+  },
+  CREATE_CHART: {
+    type: "CREATE_CHART",
+    definition: TEST_CHART_DATA.basicChart,
+    position: { x: 0, y: 0 },
+    size: { width: 200, height: 200 },
+    id: "figureId",
+    sheetId: "sheetId",
+  },
+  CREATE_IMAGE: {
+    type: "CREATE_IMAGE",
+    position: { x: 0, y: 0 },
+    size: { width: 200, height: 200 },
+    definition: { path: "/image/1", size: { width: 200, height: 200 } },
+    sheetId: "sheetId",
+    figureId: "figureId",
+  },
+  RESIZE_COLUMNS_ROWS: {
+    type: "RESIZE_COLUMNS_ROWS",
+    dimension: "ROW",
+    elements: [0],
+    size: 100,
+    sheetId: "sheetId",
+  },
+  ADD_COLUMNS_ROWS: {
+    type: "ADD_COLUMNS_ROWS",
+    position: "after",
+    dimension: "ROW",
+    base: 0,
+    quantity: 1,
+    sheetId: "sheetId",
+  },
+};
+export const TEST_COMMANDS = TEST_COMMANDS_PARTIAL as CommandMapping;
```

Comment:
this could use the `satisfies` operator introduced in Typescript 4.9
... for later

Reply:
When we have upgraded our version of TS to 4.9 I might have a look at what it does 😅